### PR TITLE
Implemeted survey response creation

### PR DESCRIPTION
**Pull request by nikolapantelic-ftn**
- created form on front-end
- implemented service for fetching surveys and creating survey responses
- created tests

### CODE REVIEW

**Comment by nikolapantelic-ftn** [comment]
[[E5.1] As a patient I want to be able to fill in a survey after an appointment, so that the hospital has a feedback of my experiences. [3]](https://trello.com/c/6fHoaL9f/27-e51-as-a-patient-i-want-to-be-able-to-fill-in-a-survey-after-an-appointment-so-that-the-hospital-has-a-feedback-of-my-experience)

**Comment by rammba** [comment]
[(0) PR Implemeted survey response creation](https://trello.com/c/6fRYku3G/92-0-pr-implemeted-survey-response-creation)

File: HospitalFrontend/src/views/CreateSurveyResponse.vue [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                    }
+                ]
+            })
+        },
+        saveSurveyResponse: function () {
+            axios.post(api.survey + '/response', this.surveyResponse)
+            .then(response => this.toastSuccess())
+            .catch(err => {
+                    this.toastError()
+                    console.log(err.response)
```
Comment (rammba) [inline HospitalFrontend/src/views/CreateSurveyResponse.vue:126]: Delete this :)

File: HospitalFrontend/src/views/CreateSurveyResponse.vue [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                            v-on:click="rateQuestion(surveySection, surveyQuestion.id, 2)">
+                        <label class="form-check-label">2</label>
+                        <input type="radio" class="form-check-input ml-2" v-bind:name="'group' + surveyQuestion.id"
+                            v-on:click="rateQuestion(surveySection, surveyQuestion.id, 3)">
+                        <label class="form-check-label">3</label>
+                        <input type="radio" class="form-check-input ml-2" v-bind:name="'group' + surveyQuestion.id"
+                            v-on:click="rateQuestion(surveySection, surveyQuestion.id, 4)">
+                        <label class="form-check-label">4</label>
+                        <input type="radio" class="form-check-input ml-2" v-bind:name="'group' + surveyQuestion.id"
+                            v-on:click="rateQuestion(surveySection, surveyQuestion.id, 5)">
```
Comment (BoKna031) [inline HospitalFrontend/src/views/CreateSurveyResponse.vue:29]: Why not use v-for to define a rate

File: HospitalFrontend/src/views/CreateSurveyResponse.vue [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        rateQuestion: function (section, questionId, rating) {
+            if (section.isDoctorSection) {
+                if (this.surveyResponse.doctorSurveySection == null) {
+                    this.surveyResponse.doctorSurveySection = {
+                        doctorId: 1,
+                        surveySectionId: section.id,
+                        ratedSurveyQuestions: [{
+                            surveyQuestionId: questionId,
+                            rating: rating
+                        }]
```
Comment (BoKna031) [inline HospitalFrontend/src/views/CreateSurveyResponse.vue:79]: I'm not sure I understand the logic behind this function.

File: HospitalFrontend/src/views/SurveyPreview.vue [hunk 4]
```diff
@@ -181,7 +181,7 @@ export default {
             if(sum==0)
                 return 0
 
-            return (rate / sum ) * 100
+            return parseFloat(((rate / sum ) * 100).toFixed(2))
```
Comment (BoKna031) [inline HospitalFrontend/src/views/SurveyPreview.vue:184]: If you divide two integer numbers as a result you will always get integer. So the result of your code will always end with .00

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 1 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue]: Deleted :)

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 2 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue:29]: I prefer using v-for for dynamic list rendering only

File: HospitalFrontend/src/views/SurveyPreview.vue (hunk 4 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/SurveyPreview.vue:184]: This is not true for javascript.

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 3 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue:79]: Unfortunately, simplification of this function would require some model changes which we intend to do in future sprints.For now, I can tell you it has been tested properly.

File: HospitalFrontend/src/views/SurveyPreview.vue (hunk 4 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/SurveyPreview.vue:184]: The way this works is: first toFixed() rounds any number, integer or float, to 2 decimals, then parseFloat() removes trailing zeros.

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 3 above)
Comment (BoKna031) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue:79]: What 'doctorId: 1' represents?

File: HospitalFrontend/src/views/CreateSurveyResponse.vue [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+    name:'CreateSurveyResponse',
+    data:function() {
+        return{          
+            survey: null,
+            surveyResponse: {
+                surveyId:null,
+                ratedSurveySections:[],
+                doctorSurveySection:null,
+                patientAccountId:1
```
Comment (BoKna031) [inline HospitalFrontend/src/views/CreateSurveyResponse.vue:59]: Why is patientAccountId initialized to 1

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 3 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue:79]: It's a hardcoded doctorId which represents a doctor about whom the survey is about. We have no forms for creating new surveys or doctors, that's why it is hardcoded right now.

File: HospitalFrontend/src/views/CreateSurveyResponse.vue (hunk 5 above)
Comment (nikolapantelic-ftn) [reply HospitalFrontend/src/views/CreateSurveyResponse.vue:59]: Right now there is no login functionality for users. This id will be fetched from session once we implement login.

**Review by BoKna031 — APPROVED** [review]

**Review by rammba — APPROVED** [review]

**Pull request merged**